src/plugins/intel_gna/frontend/layer_quantizer.hpp

```diff
@@ -105,615 +97,374 @@ struct FakeQuantizeParams {
+                                                      InferenceEngine::Precision precision,
+                                                      float scale_factor,
+                                                      const FakeQuantizeParams& fq_params) {
     InferenceEngine::Blob::Ptr result_ptr = nullptr;
     switch (precision) {
     case InferenceEngine::Precision::FP32:
-        result_ptr = fp32_to_precision_blob<float>(fp32_blob, precision, scale_factor, fqParams);
+        result_ptr = FP32ToPrecisionBlob<float>(fp32_blob, precision, scale_factor, fq_params);
         break;
     case InferenceEngine::Precision::I32:
-        result_ptr = fp32_to_precision_blob<int32_t>(fp32_blob, precision, scale_factor, fqParams);
+        result_ptr = FP32ToPrecisionBlob<int32_t>(fp32_blob, precision, scale_factor, fq_params);
         break;
     case InferenceEngine::Precision::I16:
-        result_ptr = fp32_to_precision_blob<int16_t>(fp32_blob, precision, scale_factor, fqParams);
+        result_ptr = FP32ToPrecisionBlob<int16_t>(fp32_blob, precision, scale_factor, fq_params);
         break;
     case InferenceEngine::Precision::I8:
-        result_ptr = fp32_to_precision_blob<int8_t>(fp32_blob, precision, scale_factor, fqParams);
+        result_ptr = FP32ToPrecisionBlob<int8_t>(fp32_blob, precision, scale_factor, fq_params);
         break;
     default:
         THROW_GNA_EXCEPTION << "FP32 to " << precision << " not supported";
     }
     return result_ptr;
 }
 
-template <class T, class... Args>
-InferenceEngine::Blob::Ptr make_custom_blob(Args&&... args) {
-    return InferenceEngine::make_shared_blob<T>(InferenceEngine::Precision::fromType<T>(), std::forward<Args>(args)...);
-}
-
-template <class T>
-InferenceEngine::Blob::Ptr make_custom_blob(InferenceEngine::Layout layout, InferenceEngine::SizeVector size) {
-    return InferenceEngine::make_shared_blob<T>(
-        InferenceEngine::TensorDesc(InferenceEngine::Precision::fromType<T>(), size, layout));
-}
-
-template<class QuantDesc, class QuantFunc>
-inline void quantizeWeightsBiases(const QuantDesc & quantDesc,
-                                  InferenceEngine::WeightableLayer *wl,
-                                  const QuantFunc &fnc,
-                                  bool isDiagonal = false) {  // for diagonal layer number of weights and biases significatly smaller
-    // for quantized weights
-    auto intWeights =
-        make_custom_blob<typename QuantDesc::WeightsPrecision>(InferenceEngine::C, InferenceEngine::SizeVector({wl->_weights->size()}));
-    intWeights->allocate();
-    if (intWeights->buffer() == nullptr) {
-        IE_THROW(NotAllocated)
-                << "[GNAPlugin] in function " << __PRETTY_FUNCTION__<< ": "
-                << "cannot copy weights for layer :"<< wl->name << " of size" << intWeights->byteSize();
+static size_t GetBiasSizeForLayer(InferenceEngine::WeightableLayer* wl) {
```

Comment:
```suggestion
static size_t GetBiasSizeForLayer(InferenceEngine::WeightableLayer* wl) {
```

// this will require update of code below